src/Map.php

```diff
@@ -3727,4 +3727,37 @@ protected function visit( iterable $entries, array &$result, int $level, ?\Closu
 			}
 		}
 	}
+
+
+	/**
+	 * Executes $callback if the Map is empty
+	 *
+	 * Examples: (Returns match or dies if doesn't exist)
+	 *  $page = $this->pages
+	 *          ->filter(fn(Page $page, $_) => $page->Id == $pageId)
+	 *          ->if(
+	 *              fn(Map &$Map) => $Map->isEmpty(),
+	 *              fn(Map &$_) => die("unknown page: " . $pageId))
+	 *          ->first();
+	 *
+	 * @param mixed|callable $condition Bool or function with (Map) parameter and returns bool
+	 * @param callable $then Function with (Map) parameter and returns void
+	 * @param callable|null $else Optional function with (Map) parameter and returns void
+	 * @return self Same map for fluid interface
+	 */
+	public function if($condition, callable $then, callable $else = null): self
+	{
+		$conditionResult = $condition;
+		if($condition instanceof \Closure){
+			$conditionResult = $condition($this);
+		}
+
+		if ($conditionResult) {
+			$then($this);
+		} elseif($else != null) {
+			$else($this);
+		}
```

Comment:
You can shorten this to:
```
if( $condition instanceof \Closure ? $condition( $this ) : $condition ) {
	$then( $this );
} else {
	$else( $this );
}
```

Reply:
fixed in c329afdda114015b900470c1ef73edc6439fc1a1